cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -790,6 +791,7 @@ public boolean accept(String className, URL classUrl, URL classPathUrl) {
     classes.add(MapperWrapper.class);
     classes.add(ReducerWrapper.class);
     classes.add(SLF4JBridgeHandler.class);
+    classes.add(MainClassLoader.class);
```

Comment:
It's better to have dependency on `MapReduceContainerLauncher.class` instead since the job needs to have that class to run correctly and it is that class which uses the MainClassLoader